File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/oracle/NonEmptyStringTypes.scala
package dev.typr.foundations.docs.oracle
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*



@SuppressWarnings(Array("unused"))
object NonEmptyStringTypes:
  //start
  val nonEmpty: OracleType[NonEmptyString] = OracleTypes.varchar2NonEmpty(100)
  val nvarNonEmpty: OracleType[NonEmptyString] = OracleTypes.nvarchar2NonEmpty(100)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/oracle/NonEmptyStringTypes.java
package dev.typr.foundations.docs.oracle;

import dev.typr.foundations.OracleType;
import dev.typr.foundations.OracleTypes;
import dev.typr.foundations.data.NonEmptyString;

@SuppressWarnings("unused")
public class NonEmptyStringTypes {
  // start
  OracleType<NonEmptyString> nonEmpty = OracleTypes.varchar2NonEmpty(100);
  OracleType<NonEmptyString> nvarNonEmpty = OracleTypes.nvarchar2NonEmpty(100);
  // stop
}
